### ISSUE TITLE
Bucket system issues

### ISSUE THREAD

**Issue by Buom01** [GrangerHub/tremulous#111]
Hi,
I notice that there are a persistent critical error with the new bucket system,
Step to reproduce:
1. `git clone https://github.com/GrangerHub/tremulous`
2. `make debug`
2.1. maybe you will need to patch curl path in `src/client/cl_curl.h` line 29 to compile: replace it to `#include "../../external/libcurl-7.35.0/curl/curl.h"` . (I will patch it with https://github.com/GrangerHub/tremulous/pull/110)
3. copy data-1.1.0.pk3 and data-gpp1.pk3
4. `./build/debug-linux-x86_64/tremulous`
5. Observe (with incessant restart)
```
********************
ERROR: Q_strncpyz: NULL src
********************
```

My way to found that it come from bucket system:
1. Edit `q_shared.c`, line 771
```
	if ( !src ) {
		Com_Printf("NULL src, but dest is %s\n", dest ? dest : "undefined");
		strncpy( dest, "_epicfail", destsize-1 );
		dest[destsize-1] = 0;
		// Com_Error( ERR_FATAL, "Q_strncpyz: NULL src" );
		return;
	}
```
2. `make debug && ./build/debug-linux-x86_64/tremulous`
3. Open console and type `ui_mainMenuSelectedMusic`
4. Observe
```
]/ui_mainMenuSelectedMusic
"ui_mainMenuSelectedMusic" is:"_epicfail^7" default:"^7"
```

Moreover there are some important warnings (In french, sorry):
```
CXX src/qcommon/common.cpp
src/qcommon/common.cpp: Dans la fonction « void* Z_PlaceHolderAlloc(int) »:
src/qcommon/common.cpp:3674:1: attention: « return » manquant dans une fonction devant retourner une valeur [-Wreturn-type]
 3674 | }
      | ^
src/qcommon/common.cpp: Dans la fonction « unsigned int Com_Bucket_Create_Bucket() »:
src/qcommon/common.cpp:3678:1: attention: « return » manquant dans une fonction devant retourner une valeur [-Wreturn-type]
 3678 | }
      | ^
src/qcommon/common.cpp: Dans la fonction « void* Com_Bucket_Select_A_Random_Item(unsigned int) »:
src/qcommon/common.cpp:3701:1: attention: « return » manquant dans une fonction devant retourner une valeur [-Wreturn-type]
 3701 | }
      | ^
```
Information; using the release build do the same error

**Comment by Buom01** [comment]
Closed thanks to https://github.com/GrangerHub/tremulous/pull/76/commits/798ad469103d342288ae9ab08b7d863f0d0bf002

**Comment by Buom01** [comment]
Hi,
I just noticed that https://github.com/GrangerHub/tremulous/commit/798ad469103d342288ae9ab08b7d863f0d0bf002 is still not present on 'master' branch. Is it intended ?

**Comment by dGr8LookinSparky** [comment]
It looks like I committed that on the new file system branch/pullrequest, which we didn't merge in yet (still have to do some testing server side).  I may not have time to properly test the server side aspect of the new file system for a couple more weeks.  @Buom01 , if you would like, feel free to setup a server using that branch ( https://github.com/dGr8LookinSparky/tremulous/tree/filesystem ) and let us know if you find any issues.  That would save us some time.  We have tested the client aspect of the new file system, and from what I have seen so far it works great, this is definitely something I want to get merged in.

**Comment by Buom01** [comment]
OK.
Currently I only seen that my compiled Windows binary (compiled with docker) fail silently (from server branch, with a copy of your bucket system fix that I want to remove to keep yours). It doesn't contain new FileSystem as I know.

So I tested yours binary on WineHQ this time. It seem that v1.3.0-alpha.0.14.10 works perfectly but, v1.3.0-alpha.0.14.11 fail silently too.
I hate coding on Windows, but I my current server absolutely need the client since my TeamStatus patch, so I will look deeper into common commit across theses binaries.

I'm thinking about the idea of making softer error for unhanded system call trap which could allow older client to use new QVM as currently I have some new call in my QVM and making it compatible just to allow player to play sound complicated.
But anything else like an auto-updating launcher sound better. I already started a non published project to launch all. I wait a little before continuing (like stability of Stellar-Prey, time to deep into new file system, ......)
